### PR TITLE
FINERACT-965 Upgrade, Replace itext 2.1.7 with Openpdf 1.1.0

### PR DESCRIPTION
**Pull request by nnatarajan**
FINERACT-965

### CODE REVIEW

File: fineract-provider/build.gradle [hunk 1]
```diff
@@ -87,8 +87,9 @@ dependencyManagement {
         dependency 'org.apache.commons:commons-email:1.5'
         dependency 'org.apache.commons:commons-io:+'
         dependency 'org.drizzle.jdbc:drizzle-jdbc:1.4'
-        dependency 'com.lowagie:itext:2.1.7'
-        dependency 'com.lowagie:itext-rtf:2.1.7'
+        dependency 'com.lowagie:itext:4.2.2'
+        dependency 'com.lowagie:itext-rtf:4.2.2'
+        dependency 'com.itextpdf:itextpdf:5.5.13'
```
Comment (vorburger) [inline fineract-provider/build.gradle]: The mix of v4 and v5 here seems curious... is this really right?

**Review by vorburger — CHANGES_REQUESTED** [review]

File: fineract-provider/build.gradle (hunk 1 above)
Comment (nnatarajan) [reply fineract-provider/build.gradle]: When I updated itext, I got error, requiring com.itextpdf:itextpdf. It seems that they are independent vendors. Version 5 is the most current version and dates back to 2011.

Comment (nnatarajan) [reply fineract-provider/build.gradle]: https://mvnrepository.com/artifact/com.itextpdf/itextpdf

**Comment by vorburger** [comment]
Wait a sec, so iText v5 is AGPL licensed?! Then we can't use it in Fineract, see https://www.apache.org/legal/resolved.html#category-x ...

Reading https://en.m.wikipedia.org/wiki/IText and seeing on https://search.maven.org/search?q=iText that v4.2.2 is ancient from 2015.

I suggest that we use this as the opportunity to switch to https://github.com/LibrePDF/OpenPDF instead. That's dual LGPL (NOK) and MPL (fine).

**Comment by nnatarajan** [comment]
> Wait a sec, so iText v5 is AGPL licensed?! Then we can't use it in Fineract, see https://www.apache.org/legal/resolved.html#category-x ...
> 
> Reading https://en.m.wikipedia.org/wiki/IText and seeing on https://search.maven.org/search?q=iText that v4.2.2 is ancient from 2015.
> 
> I suggest that we use this as the opportunity to switch to https://github.com/LibrePDF/OpenPDF instead. That's dual LGPL (NOK) and MPL (fine).

Is the latest version ok -- OpenPDF 1.3.17 ?

**Comment by nnatarajan** [comment]
> Wait a sec, so iText v5 is AGPL licensed?! Then we can't use it in Fineract, see https://www.apache.org/legal/resolved.html#category-x ...
> 
> Reading https://en.m.wikipedia.org/wiki/IText and seeing on https://search.maven.org/search?q=iText that v4.2.2 is ancient from 2015.
> 
> I suggest that we use this as the opportunity to switch to https://github.com/LibrePDF/OpenPDF instead. That's dual LGPL (NOK) and MPL (fine).

Dependency Openpdf: 1.1.0 is the most recent version to fit all class requirements.

**Review by vorburger — CHANGES_REQUESTED** [review]
> Dependency Openpdf: 1.1.0 is the most recent version to fit all class requirements.

https://github.com/LibrePDF/OpenPDF says latest is 1.3.17 ... does that not "fit all class requirements"? Can you elaborate on what you mean by that? It should, hopefully, be possible to upgrade to the latest, even if some import statements or some code has to be changed...

**Comment by nnatarajan** [comment]
> > Dependency Openpdf: 1.1.0 is the most recent version to fit all class requirements.
> 
> https://github.com/LibrePDF/OpenPDF says latest is 1.3.17 ... does that not "fit all class requirements"? Can you elaborate on what you mean by that? It should, hopefully, be possible to upgrade to the latest, even if some import statements or some code has to be changed...

@vorburger 
OpenPDF: 1.1.0 from 2018 is the latest version, which will recognize this import in a few classes -- com.lowagie.text.pdf.codec.Base64.

More recent versions of OpenPDF do not recognize Base64

**Comment by vorburger** [comment]
> More recent versions of OpenPDF do not recognize Base64

thanks for having identified that. FYI Base64 is a (very common) "encoding scheme", read https://en.wikipedia.org/wiki/Base64, and as per http://www.docjar.com/docs/api/com/lowagie/text/pdf/codec/Base64.html => http://iharder.sourceforge.net/current/java/base64/ they just ditched that because now there is https://docs.oracle.com/javase/8/docs/api/java/util/Base64.html

I've done the "conversion" in new #948 (that PR includes your commit, for correct "attribution").

OK to close this?

**Comment by nnatarajan** [comment]
> > More recent versions of OpenPDF do not recognize Base64
> 
> thanks for having identified that. FYI Base64 is a (very common) "encoding scheme", read https://en.wikipedia.org/wiki/Base64, and as per http://www.docjar.com/docs/api/com/lowagie/text/pdf/codec/Base64.html => http://iharder.sourceforge.net/current/java/base64/ they just ditched that because now there is https://docs.oracle.com/javase/8/docs/api/java/util/Base64.html
> 
> I've done the "conversion" in new #948 (that PR includes your commit, for correct "attribution").
> 
> OK to close this?

@vorburger  Ok.

**Pull request closed without merging**